TF_Service_dotNet/TouchFree/InteractionManager.cs

```diff
@@ -0,0 +1,177 @@
+
+            if (_config.InteractionType == InteractionType.PUSH)
+            {
+                interactionsToUse.Add(InteractionType.PUSH);
+
+                if (interactionTuning?.EnableAirClickWithAirPush == true)
+                {
+                    interactionsToUse.Add(InteractionType.AIRCLICK);
+                }
+
+                if (interactionTuning?.EnableVelocitySwipeWithAirPush == true)
+                {
+                    interactionsToUse.Add(InteractionType.VELOCITYSWIPE);
+                }
+            }
+            else
+            {
+                interactionsToUse.Add(_config.InteractionType);
+            }
+
+            activeInteractions = interactions.Where(x => interactionsToUse.Contains(x.InteractionType)).ToDictionary(x => x, x => 1f);
+            locationInteraction = interactions.SingleOrDefault(x => x.InteractionType == _config.InteractionType);
+
+            if (initialisationNotStarted)
+            {
+                updateBehaviour.OnUpdate += Update;
+            }
+        }
+
+        protected void Update()
+        {
+            if (activeInteractions != null)
+            {
+                InputAction? inputAction = null;
+                float currentMaxProgress = 0;
+                InputAction? lastLocationActionToUpdate = null;
+
+                if (interactionCurrentlyDown != null)
+                {
+                    var interactionInputAction = interactionCurrentlyDown.Update(1);
+                    inputAction = interactionInputAction.inputAction;
+                    currentMaxProgress = inputAction?.ProgressToClick ?? 0;
+
+                    if (!inputAction.HasValue || inputAction.Value.InputType == InputType.UP || inputAction.Value.InputType == InputType.CANCEL)
+                    {
+                        if (interactionCurrentlyDown != locationInteraction)
+                        {
+                            lastLocationActionToUpdate = locationInteraction.Update(1).inputAction;
+                        }
+                        interactionCurrentlyDown = null;
+                    }
+                }
+                else
+                {
+                    foreach (var interaction in activeInteractions)
+                    {
+                        var interactionInputAction = interaction.Key.Update(interaction.Value);
+
+                        if (interaction.Key == locationInteraction)
+                        {
+                            lastLocationActionToUpdate = interactionInputAction.inputAction;
+                            if (interactionCurrentlyDown == null)
+                            { 
+                                inputAction = interactionInputAction.inputAction;
+                            }
+                        }
+
+                        if (interactionCurrentlyDown == null && interactionInputAction != null && interactionInputAction.actionDetected && interactionInputAction.inputAction.InputType == InputType.DOWN)
```

Comment:
I don't think that this is a very big issue as it would require them both hitting down on the same frame. I was also thinking that we would add the interactions into the dictionary in priority order

Reply:
I was wondering if we could trivially sort the list by the weight (if using weighted interactions) as well to lean in favour of this, but yes you're right because down happens on a _single frame_ this is very unlikely to occur. 👍